dev/com.ibm.ws.org.apache.cxf.cxf.rt.databinding.jaxb.3.2/src/org/apache/cxf/jaxb/JAXBDataBinding.java

```diff
@@ -426,24 +470,43 @@ public synchronized void initialize(Service service) {
                 }
             }
             for (DOMSource r : schemas) {
+                if (LOG.isLoggable(Level.FINE)) { // Liberty change begin
+                   LOG.fine("initialize: Processing DOMSource node: " + 
+				(r != null ? StaxUtils.toString(r.getNode()) : "null"));
+                } // Liberty change end
                 if (bi.contains(r)) {
                     String ns = ((Document)r.getNode()).getDocumentElement().getAttribute("targetNamespace");
+                    if (LOG.isLoggable(Level.FINE)) { // Liberty change begin
+                       LOG.fine("initialize: BuiltIn schema contains DOMSource r with NS: " + ns);
+                    }  // Liberty change end
                     if (serviceInfo.getSchema(ns) != null) {
+                        if (LOG.isLoggable(Level.FINE)) { // Liberty change begin
+                           LOG.fine("initialize: serviceInfo has schema, continue");
+                        }  // Liberty change end
                         continue;
                     }
                 }
                 //StaxUtils.print(r.getNode());
                 //System.out.println();
+                if (LOG.isLoggable(Level.FINE)) { // Liberty change begin
+                   LOG.fine("initialize: Adding SchemaDocument for: " + r.getSystemId());
+                } // Liberty change end
                 addSchemaDocument(serviceInfo,
                                   col,
                                  (Document)r.getNode(),
                                   r.getSystemId());
             }
 
+            if (LOG.isLoggable(Level.FINE)) { // Liberty change begin
```

Comment:
Maybe continue using isLoggableFine instead of checking again?

Reply:
yes, will update